modules/core/src/logger.cpp

```diff
@@ -233,24 +233,42 @@ void writeLogMessage(LogLevel logLevel, const char* message)
         (*out) << std::flush;
 }
 
+static const char* stripSourceFilePathPrefix(const char* file)
+{
+    std::string file_s(file);
+    size_t pos1 = file_s.rfind('/');
+    size_t pos2 = file_s.rfind('\\');
+    size_t pos = 0;
+    if (pos2 == std::string::npos)
+        pos = pos1;
+    else if (pos1 == std::string::npos)
+        pos = pos2;
+    else
+        pos = std::max(pos1, pos2);
+    if (pos == std::string::npos || pos >= file_s.length() - 1)
+        return file;
+    return file + pos + 1;
```

Comment:
What if we use C for this function? Something like:

    const char * pos = std::max(strrchr(file, '/'), strrchr(file, '\\'));
    if (!pos)
        return file;
    return pos + 1;

It will be inlined and evaluated at compile time unlike C++.

Reply:
Replaced C++ verbose code.

Reply:
Interesting, for some reason only clang (5+) is able to evaluate this function at compile time, even the latest GCC adds an inlined function call: https://godbolt.org/z/v8PE7K85h While the variant with `strrchr` is consistently replaced with a precomputed pointer by older compilers such as GCC 4.6.4 and clang 3.6.

Also I just thought that in order to enable compile time evaluation we need to directly wrap `__FILE__` macro here: https://github.com/opencv/opencv/blob/ab912329b67d3a142a1b399b4bc9fa9b68cd8df6/modules/core/include/opencv2/core/utils/logger.hpp#L122 On the other hand if we do so then in builds without optimization (`-O0`) each log message will add this function call.

Perhaps I'm digging too deep here and the simple readable code is fine in the end.